### PR TITLE
add stale read test

### PR DESCRIPTION
**Pull request by NingLin-P**
### Check List 

Tests 

 - Unit test
 - E2E test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Go code change
 - Has CI related scripts change
 - Has Terraform scripts change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: testcase/stale-read/client.go [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+
+// NextRequest
+func (s *staleRead) NextRequest() interface{} {
+	panic("implement me")
+}
+
+// DumpState
+func (s *staleRead) DumpState(ctx context.Context) (interface{}, error) {
+	panic("implement me")
+}
```
Comment (mahjonp) [inline testcase/stale-read/client.go:319]: All these methods aren't necessary, you can delete them.

**Review by mahjonp — COMMENTED** [review]
Rest LGTM, I just take a brief look

File: testcase/stale-read/client.go [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+			return
+		default:
+			time.Sleep(s.RequestInterval)
+			count++
+		}
+	}
+}
+
+func (s *staleRead) staleRead(ctx context.Context, conn *sql.Conn, t time.Time, ids []int) (map[int]string, uint64, error) {
+	txn, err := conn.BeginTx(ctx, &sql.TxOptions{})
```
Comment (Yisaer) [inline testcase/stale-read/client.go:194]: `BeginTx` here will directly send `begin` to sql server. I think we should use `ExecuteContext` to begin a stale transaction by `conn.BeginTx(ctx, "START TRANSACTION READ ONLY AS OF TIMESTAMP...") `

**Review by Yisaer — COMMENTED** [review]
REST lgtm

File: testcase/stale-read/client.go (hunk 2 above)
Comment (NingLin-P) [reply testcase/stale-read/client.go]: `conn.BeginTx(ctx, "START TRANSACTION READ ONLY AS OF TIMESTAMP...")` not match the function signature `func (*sql.Conn).BeginTx(ctx context.Context, opts *sql.TxOptions)`, I use the `SET TRANSACTION READ ONLY AS OF TIMESTAMP` instead, PTAL

**Review by Yisaer — APPROVED** [review]

**Pull request merged**